airflow/models/skipmixin.py

```diff
@@ -143,7 +146,12 @@ def skip(
                 session=session,
             )
 
-    def skip_all_except(self, ti: TaskInstance, branch_task_ids: None | str | Iterable[str]):
+    def skip_all_except(
+        self,
+        ti: TaskInstance | TaskInstancePydantic,
+        task: BaseOperator,
```

Comment:
Hmm, this feels like it could be a breaking change? Why do we need to do this?

Reply:
Yeah You are likely right, I will bring it back.

The problem is that task attribute cannot be **really** defined in TaskInstancePydantic and we should not be serialized - becuase it is only set and used during execution preparation. So  rather than retrieving and serializing it fro from the DB it is populated during execution of the task in local task job and I could not find a good way to add dynamically a non-serializable attribute to TaskInstancePydantic.

But this part will be handled later when we extract the methods from task_instance to be AIP-44 compliant, so we can handle it then. For now I will add # type: ignore to handle it.

Reply:
Yep. Much better when I returned it I think.